src/plugins/ui_actions/public/tests/get_trigger_compatible_actions.test.ts

```diff
@@ -59,14 +59,14 @@ test('getTriggerCompatibleActions returns attached actions', async () => {
 
 test('filters out actions not applicable based on the context', async () => {
   const { setup, doStart } = uiActions;
-  const action1 = createAction({
```

Comment:
Looks like there is more mix up now than before (in tests) between `Action` which is registered and returned by the registry and `ActionDefinition` which is the object that is used to register an action. 

Not sure if there was a significant reason to mixing up them more than before, but I'd prefer we'd not make it worse if possible. I'd suggest we try to keep them clear and separate and not try to reuse a single object for a definition and the registry item via type cast.

Reply:
I would like to get rid of `createAction` and `Action` eventually, those are legacy code artefacts.

I'm not sure what you mean about code mixing. Before this PR a bunch of places register in the registry `Action`, which is incorrect, because `ActionDefinition` should be registered. Part of this PR cleans up a bunch of these places which register `Action`.

Reply:
Or maybe we should leave `Action`. (I still would like to get rid of `createAction`.)